src/core/structure/sheet.ts

```diff
@@ -58,24 +58,30 @@ export default class Sheet {
 
   setCellAt(colPos: number, rowPos: number, value: string): CellInfo {
     const position = this.initializePosition(colPos, rowPos);
-    return this.setCell(position.columnKey!, position.rowKey!, value);
+    return this.setCell(position.columnKey!, position.rowKey!, value)!;
   }
 
-  setCell(colKey: ColumnKey, rowKey: RowKey, formula: string): CellInfo {
+  setCell(colKey: ColumnKey, rowKey: RowKey, formula: string): CellInfo | null {
+    if (!this.columns.has(colKey) || !this.rows.has(rowKey)) {
+      return null;
+    }
```

Comment:
Since we have this check here, there are redundant checks in the return statement. And I don't think rowKey and columnKey need to be nullable on PositionInfo type

Reply:
The checks are not redundant. The `deleteCellIfUnused` call can lead to the column or row being deleted. This is also why the keys are nullable in `PositionInfo`.